tcmc/tcmc.py

```diff
@@ -100,20 +104,35 @@ def __parse_forest(self,forest):
         self._initial_lengths = lengths
 
     def build(self, input_shape):
-        
+
         s = input_shape[-1]
         self.alphabet_size = s
-        
+
         M = np.prod(self.model_shape)
         
         rates_initializer = self.rates_initializer if self.rates_initializer != None else tf.initializers.RandomUniform(minval=-1, maxval=1)
         stationary_distribution_initializer = self.stationary_distribution_initializer if self.stationary_distribution_initializer != None else tf.initializers.constant(1.0 / (np.sqrt(s) - 1))
         
-        
-        # The parameters that we want to learn
-        self.R_inv = self.add_weight(shape = (M, int(s*(s-1)/2)), name = "R_inv", dtype = tf.float64,
-                                     initializer = rates_initializer)
-        
+        if not self.sparse_rates:
+            # The parameters that we want to learn
+            self.R_inv = self.add_weight(shape = (M, int(s*(s-1)/2)), name = "R_inv", dtype = tf.float64,
+                                         initializer = rates_initializer)
+        else:
+            max_tuple_length = 10
+            nuc_s = [4 ** i for i in range(1, max_tuple_length)]
```

Comment:
add some comment here, like currently for dna (4) and amino acids (20) alphabets only
Is t initialized somewhere?
What happens when
 1. the tuple length is larger than `max_tuple_length`? - at least a comment
 2. when `s` is not found in either set?

Reply:
Is t initialized somewhere? - No. Only s is initialized.